code/modules/shuttle/special.dm

```diff
@@ -162,10 +166,20 @@
 	unique_name = FALSE // disables the (123) number suffix
 	initial_language_holder = /datum/language_holder/universal
 	default_storage = null
+	///The roundend callback tied to the drone that we use to remove its pacification.
+	var/datum/callback/roundend_callback
+
+/mob/living/basic/drone/snowflake/bardrone/Destroy()
+	LAZYREMOVE(SSticker.round_end_events, roundend_callback)
+	roundend_callback = null
+	return ..()
 
 /mob/living/basic/drone/snowflake/bardrone/Initialize(mapload)
 	. = ..()
 	AddComponentFrom(ROUNDSTART_TRAIT, /datum/component/area_based_godmode, area_type = /area/shuttle/escape, allow_area_subtypes = TRUE)
+	ADD_TRAIT(src, TRAIT_PACIFISM, INNATE_TRAIT) //drone is pacified
+	roundend_callback = TRAIT_CALLBACK_REMOVE(src, TRAIT_PACIFISM, INNATE_TRAIT)
```

Comment:
As I said in my previous comments, it's one or the other. If you want to use callback remove instead, then get rid of ``round_end_depacify`` proc cause it's not used anymore.

Reply:
if you really want the balloon alert you can do something really funny with `RegisterSignal(src, SIGNAL_REMOVETRAIT(TRAIT_PACIFISM), PROC_REF(round_end_depacify))` and remove the `REMOVE_TRAIT()` call and add a `SIGNAL_HANDLER` to the top of `round_end_depacify()`

Reply:
basically

```dm
/mob/barmaid/Initialize(...)
 ADD_TRAIT(src, TRAIT_PACIFISM, INNATE_TRAIT)
 roundend_callback = TRAIT_CALLBACK_REMOVE(src, TRAIT_PACIFISM, INNATE_TRAIT) // we already clean this up on Destroy()
 RegisterSignal(src, SIGNAL_REMOVETRAIT(TRAIT_PACIFISM), PROC_REF(round_end_depacify))

/mob/barmaid/proc/round_end_depacify()
 SIGNAL_HANDLER
 balloon_alert(src, "no longer pacified!")
```